fhevm/params.go

```diff
@@ -214,31 +223,38 @@ func DefaultGasCosts() GasCosts {
 			tfhe.FheUint64: 43000 + AdjustFHEGas,
 		},
 		FheGetCiphertext: map[tfhe.FheUintType]uint64{
-			tfhe.FheUint8:  12000,
-			tfhe.FheUint16: 14000,
-			tfhe.FheUint32: 18000,
-			tfhe.FheUint64: 28000,
+			tfhe.FheBool:     10000,
+			tfhe.FheUint8:    12000,
+			tfhe.FheUint16:   14000,
+			tfhe.FheUint32:   18000,
+			tfhe.FheUint64:   28000,
+			tfhe.FheUint160:  50000,
+			tfhe.FheUint2048: 100000,
 		},
 		// TODO: The values here are chosen somewhat arbitrarily.
 		// Also, we don't take into account whether a ciphertext existed (either "current" or "original") for the given handle.
 		// Finally, costs are likely to change in the future.
 		FheStorageSstoreGas: map[tfhe.FheUintType]uint64{
-			tfhe.FheUint4:   SstoreFheUint4Gas,
-			tfhe.FheUint8:   SstoreFheUint4Gas * 2,
-			tfhe.FheUint16:  SstoreFheUint4Gas * 4,
-			tfhe.FheUint32:  SstoreFheUint4Gas * 8,
-			tfhe.FheUint64:  SstoreFheUint4Gas * 16,
-			tfhe.FheUint128: SstoreFheUint4Gas * 32,
-			tfhe.FheUint160: SstoreFheUint4Gas * 40,
+			tfhe.FheBool:     SstoreFheUint4Gas / 2,
```

Comment:
Shouldn't price be divided by 4 instead of 2, since ebool is a single bit?

Reply:
Let me check, because it is related to serialised size and not plaintext bits. Overall, these values might be off.

Reply:
@jatZama Sizes for these data types as serialised:

```

FheBool (0) = 0x0:
8268 = 0x204c
FheUint4 (1) = 0x0:
16544 = 0x40a0
```

Reply:
weird... wondering why ebool is an exception here in respect to size.

Reply:
Because of the way the ciphetext is represented, I think. I guess 4 bit is two 2 bit low-level ciphertexts. And 1 bit is just one.